### PR TITLE
Add CI, Make tests work, Deprecate EOL releases, Update Deps

### LINKED ISSUE #235: CI update and drop old versions
I see the code is still supporting old versions of Kafka.

Looking at [this page](https://endoflife.date/apache-kafka), I see that we can safely drop versions < 3, and focus on the supported versions.

What do you think?

Tasks:

- Add Github actions to only support v3 and higher
- Fix tests to support v3 and higher

### PR DESCRIPTION
**Pull request by oneslash**
Here what I have done:

- Clean up doc, remove the Travis CI yaml, and the badge
- Add CI and make tests work for secure step
- Update deps
- Remove unsupported versions, only support 3.4 and up
- Removed image with Rust logo, to be safe that Rust Foundation doesn't come after this repository

Also closes #235

### CODE REVIEW

**Comment by oneslash** [comment]
It will be also nice to merge the following PR to update docs: https://github.com/kafka-rust/kafka-rust/pull/227

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+name: Rust
+
+on:
+  push:
+    branches: [ "main" ]
```
Comment (oneslash) [inline .github/workflows/rust.yml:5]: I need to make it for master branch too

File: tests/kafka-rust-image/create-topics.sh [hunk 2]
```diff
@@ -1,6 +1,8 @@
 #!/bin/bash
 
+set -x
 
+SECURE_PARAM=""
```
Comment (oneslash) [inline tests/kafka-rust-image/create-topics.sh:5]: Remove unused

File: tests/run-all-tests [hunk 3]
```diff
@@ -18,7 +18,7 @@
 # `wurstmeister/kafka` repo. If there are no versions specified, it runs the tests
 # on a set of default versions.
 
-set -e
+set -e -x
```
Comment (oneslash) [inline tests/run-all-tests:21]: Remove -x

**Comment by oneslash** [comment]
Closing as I want to continue my `main` branch. I created  #238 in case some day the maintainer will merge it.

**Comment by johnward** [comment]
Done, thank for your help.

**Pull request closed without merging**